datafusion/optimizer/src/utils.rs

```diff
@@ -276,9 +276,13 @@ pub fn rewrite_expression(expr: &Expr, expressions: &[Expr]) -> Result<Expr> {
         }
         Expr::Not(_) => Ok(Expr::Not(Box::new(expressions[0].clone()))),
         Expr::Negative(_) => Ok(Expr::Negative(Box::new(expressions[0].clone()))),
+        Expr::InList { list, negated, .. } => Ok(Expr::InList {
+            expr: Box::new(expressions[0].clone()),
+            list: list.clone(),
+            negated: *negated,
+        }),
         Expr::Column(_)
         | Expr::Literal(_)
-        | Expr::InList { .. }
         | Expr::Exists { .. }
```

Comment:
🤔  looks like `Exists` and `Subquery` may need the same treatment (not in this PR though).... I'll file a follow on

Reply:
Agree!